File: telegram-bot-spring-boot-starter/src/main/java/ru/tggc/telegrambotspringbootstarter/autoconfigure/LifecycleAutoConfiguration.java
package ru.tggc.telegrambotspringbootstarter.autoconfigure;

import org.springframework.boot.autoconfigure.AutoConfigureAfter;
import org.springframework.context.annotation.Bean;
import ru.tggc.telegrambotspringbootstarter.TelegramLifecycle;
import ru.tggc.telegrambotspringbootstarter.runner.TelegramBotRunner;

@AutoConfigureAfter(TelegramBotAutoConfiguration.class)
public class LifecycleAutoConfiguration {

    @Bean
    public TelegramLifecycle telegramLifecycle(TelegramBotRunner telegramBotRunner) {
        return new TelegramLifecycle(telegramBotRunner);
    }
}

File: telegram-bot-spring-boot-starter/src/main/kotlin/ru/tggc/telegrambotspringbootstarter/autoconfigure/LifecycleAutoConfiguration.kt
package ru.tggc.telegrambotspringbootstarter.autoconfigure

import org.springframework.boot.autoconfigure.AutoConfigureAfter
import org.springframework.context.annotation.Bean
import ru.tggc.telegrambotspringbootstarter.TelegramLifecycle
import ru.tggc.telegrambotspringbootstarter.runner.TelegramBotRunner

@AutoConfigureAfter(TelegramBotAutoConfiguration::class)
class LifecycleAutoConfiguration {

    @Bean
    fun telegramLifecycle(telegramBotRunner: TelegramBotRunner): TelegramLifecycle =
        TelegramLifecycle(telegramBotRunner)

}
